src/sss_client/sss_pam_compat.h

```diff
@@ -49,4 +49,15 @@
 # define PAM_BAD_ITEM PAM_USER_UNKNOWN
 #endif /* PAM_BAD_ITEM */
 
+#if !defined(PAM_NONNULL) && defined(OPENPAM_NONNULL)
+#define PAM_NONNULL OPENPAM_NONNULL
+#endif
+
+#ifndef HAVE_SECURITY_PAM_MODUTIL_H
```

Comment:
We are already checking for pam_modutil_getlogin availability during [configuration](https://github.com/SSSD/sssd/blob/master/src/external/pam.m4#L26) so I think you should check for `HAVE_PAM_MODUTIL_GETLOGIN` instead of `HAVE_SECURITY_PAM_MODUTIL_H`, as this will be more specific.